grammars/css.cson

```diff
@@ -20,294 +20,1176 @@
+            'name': 'entity.other.attribute-name.pseudo-class.css'
+          '2':
+            'name': 'punctuation.definition.entity.css'
+          '3':
+            'name': 'punctuation.section.function.begin.bracket.round.css'
+        'end': '\\)'
+        'endCaptures':
+          '0':
+            'name': 'punctuation.section.function.end.bracket.round.css'
         'patterns': [
           {
-            'include': '$self'
+            'match': '(?i)[+-]?(\\d+n?|n)(\\s*[+-]\\s*\\d+)?'
+            'name': 'constant.numeric.css'
+          }
+          {
+            'match': '(?i)even|odd'
+            'name': 'support.constant.parity.css'
           }
         ]
       }
     ]
-  }
-  {
-    'begin': '(@)page\\b'
-    'captures':
-      '0':
-        'name': 'keyword.control.at-rule.page.css'
-      '1':
-        'name': 'punctuation.definition.keyword.css'
-    'end': '\\s*((?=[{:]|$))'
-    'name': 'meta.at-rule.page.css'
-    'patterns': [
-      {
-        'include': '#rule-list'
-      }
-    ]
-  }
-  {
-    'begin': '((@)namespace\\b)\\s*(?=url)'
-    'captures':
-      '1':
-        'name': 'keyword.control.at-rule.namespace.css'
-      '2':
-        'name': 'punctuation.definition.keyword.css'
-      '3':
-        'name': 'support.function.misc.css'
-    'comment': 'Namespace without prefix'
-    'end': '\\s*((?=;|$))'
-    'name': 'meta.at-rule.namespace.css'
-    'patterns': [
-      {
-        'include': '#property-values'
-      }
-      {
-        'include': '#string-double'
-      }
-      {
-        'include': '#string-single'
-      }
-    ]
-  }
-  {
-    'begin': '((@)namespace) ([\\w-]*)'
+  'media-features':
     'captures':
       '1':
-        'name': 'keyword.control.at-rule.namespace.css'
+        'name': 'support.type.property-name.media.css'
       '2':
-        'name': 'punctuation.definition.keyword.css'
+        'name': 'invalid.deprecated.media.css'
       '3':
-        'name': 'entity.name.function.css'
-    'comment': 'Namespace'
-    'end': '\\s*((?=;|$))'
-    'name': 'meta.at-rule.namespace.css'
+        'name': 'support.type.vendored.property-name.media.css'
+    'match': '''(?xi)
+      (?<=^|\\s|\\(|\\*/)           # Preceded by whitespace, bracket or comment
+
+      (?:
+        # Standardised features
+        (
+          (?:min-|max-)?
+          (?: height
+            | width
+            | aspect-ratio
+            | color
+            | color-index
+            | monochrome
+            | resolution
+          )
+          | grid
+          | scan
+          | orientation
+          | display-mode
+        )
+
+        |
+
+        # Deprecated features
+        (
+          (?:min-|max-)?            # Deprecated in Media Queries 4
+          device-
+          (?: height
+            | width
+            | aspect-ratio
+          )
+        )
+
+        |
+
+        # Vendor extensions
+        (
+          (?:
+            # Spec-compliant syntax
+            [-_]
```

Comment:
Yes, but I'm reviewing the whole diff at once, which doesn't show the individual commit messages.  I don't review individual commits because many times I comment on something just for the next commit to have fixed it.  But yes, I could have probably looked at which commit added this in a separate tab to see if it had some extra context associated with it.